### PR TITLE
Support sum in xapi.

### PR DESCRIPTION
**Pull request by shenli**
Support sum in local_region.
Please note that there is a small change in DecodeDecimal. See the comment for the detail.
@coocood @zimulala @BusyJay

### CODE REVIEW

File: executor/aggregate_xapi_test.go [hunk 1]
```diff
@@ -178,3 +179,84 @@ func (s *testAggFuncSuite) TestXAPIFirstRow(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(val, testutil.DatumEquals, types.NewDatum(nil))
 }
+
+func (s *testAggFuncSuite) TestXAPISum(c *C) {
+	defer testleak.AfterTest(c)()
+	// Compose aggregate exec for "select c2 from t groupby c1";
```
Comment (coocood) [inline executor/aggregate_xapi_test.go]: The comment for this test is confusing.

File: store/localstore/local_aggregate.go [hunk 2]
```diff
@@ -102,18 +103,32 @@ func (n *aggregateFuncExpr) update(ctx *selectContext, args []types.Datum) error
...
 		return n.updateFirst(ctx, args)
+	case tipb.ExprType_Sum:
+		return n.updateSum(ctx, args)
 	}
 	return errors.Errorf("Unknown AggExpr: %v", n.expr.GetTp())
 }
 
-func (n *aggregateFuncExpr) toDatums() []types.Datum {
+func (n *aggregateFuncExpr) toDatums() (ds []types.Datum, err error) {
+
```
Comment (zimulala) [inline store/localstore/local_aggregate.go]: Remove this empty line.

**Comment by coocood** [comment]
LGTM

File: executor/aggregate_xapi_test.go [hunk 3]
```diff
@@ -178,3 +179,98 @@ func (s *testAggFuncSuite) TestXAPIFirstRow(c *C) {
...
+	agg := &XAggregateExec{
+		AggFuncs: []*ast.AggregateFuncExpr{fc},
+		Src:      src,
+	}
+	ast.SetFlag(fc)
+	var (
+		row *Row
+		err error
+	)
+	// First Group: 15
```
Comment (zimulala) [inline executor/aggregate_xapi_test.go]: Group -> Row ?

File: executor/aggregate_xapi_test.go [hunk 4]
```diff
@@ -178,3 +179,98 @@ func (s *testAggFuncSuite) TestXAPIFirstRow(c *C) {
...
+	)
+	// First row: 15
+	row, err = agg.Next()
+	c.Assert(err, IsNil)
+	c.Assert(row, NotNil)
+	ctx := mock.NewContext()
+	val, err := evaluator.Eval(ctx, fc)
+	c.Assert(err, IsNil)
+	c.Assert(val, testutil.DatumEquals, types.NewDecimalDatum(mysql.NewDecimalFromInt(int64(15), 0)))
+	// Second Row: 21
```
Comment (zimulala) [inline executor/aggregate_xapi_test.go]: Row -> row here or row -> Row in line 249.

**Comment by zimulala** [comment]
Rest LGTM

**Comment by shenli** [comment]
PTAL @zimulala

**Comment by zimulala** [comment]
LGTM

File: store/localstore/local_aggregate.go [hunk 5]
```diff
@@ -168,3 +183,25 @@ func (n *aggregateFuncExpr) updateFirst(ctx *selectContext, args []types.Datum)
 	aggItem.evaluated = true
 	return nil
 }
+
+func (n *aggregateFuncExpr) updateSum(ctx *selectContext, args []types.Datum) error {
+	for _, a := range args {
```
Comment (shenli) [inline store/localstore/local_aggregate.go]: I think it should not be larger than 1. I will update the code.

File: store/localstore/local_aggregate.go [hunk 6]
```diff
@@ -168,3 +183,25 @@ func (n *aggregateFuncExpr) updateFirst(ctx *selectContext, args []types.Datum)
...
+		aggItem.evaluated = true
+		aggItem.count = 1
+		return nil
+	}
+	var err error
+	aggItem.value, err = xeval.ComputeArithmetic(tipb.ExprType_Plus, args[0], aggItem.value)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	aggItem.count++
```
Comment (BusyJay) [inline store/localstore/local_aggregate.go]: `count` seems unnecessary.

Comment (shenli) [inline store/localstore/local_aggregate.go]: I will use the same function to implement avg. It will not cost too much.

**Pull request merged**